### PR TITLE
#10900 This commit includes:

### PR DESCRIPTION
**Pull request by nollymar**
- Better exception handling when an assertion is null and/or attributes can not be got from the assertion
- Docs updates
- DotSamlException and AttributesNotFoundException were moved to com.dotcms.plugin.saml.v3.exception

### CODE REVIEW

File: src/com/dotcms/plugin/saml/v3/OpenSamlAuthenticationServiceImpl.java [hunk 1]
```diff
@@ -189,6 +190,10 @@ private AttributesBean resolveAttributes (final Assertion assertion, final Confi
 
         final AttributesBean.Builder attrBuilder = new AttributesBean.Builder();
 
+        if (validateAttributes(assertion)){
+            throw new AttributesNotFoundException("No attributes found");
+        }
+
```
Comment (dsilvam) [inline src/com/dotcms/plugin/saml/v3/OpenSamlAuthenticationServiceImpl.java]: Minor thing: let's have the validateAttributes throw the exception and replace these three lines with just one call to validateAttributes

File: src/com/dotcms/plugin/saml/v3/OpenSamlAuthenticationServiceImpl.java [hunk 2]
```diff
@@ -212,23 +217,32 @@ private AttributesBean resolveAttributes (final Assertion assertion, final Confi
         return attrBuilder.build();
     } // resolveAttributes.
 
+    private boolean validateAttributes(Assertion assertion) {
```
Comment (dsilvam) [inline src/com/dotcms/plugin/saml/v3/OpenSamlAuthenticationServiceImpl.java]: add "throws AttributesNotFoundException" to signature

Comment (dsilvam) [inline src/com/dotcms/plugin/saml/v3/OpenSamlAuthenticationServiceImpl.java]: change return type to void

**Comment by nollymar** [comment]
The changes were applied

**Pull request merged**